### PR TITLE
Add 'wrapdateline' to Geometry.to_crs

### PR DESCRIPTION
**Pull request by v0lat1le**
Adding support for coordinate systems spanning dateline

### CODE REVIEW

File: datacube/utils/geometry.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def buffer(self, distance, quadsecs=30):
     def simplify(self, tolerance):
         return _make_geom_from_ogr(self._geom.Simplify(tolerance), self.crs)
 
-    def to_crs(self, crs, resolution=None):
+    def to_crs(self, crs, resolution=None, wrapdateline=False):
```
Comment (omad) [inline datacube/utils/geometry.py:425]: It'd be good to get some documentation here describing what this does.

Mention that it only makes sense for geographic (maybe even throw an exception if passing it for a non-geographic projection.) And I assume that it turns a Polygon into a MultiPolygon with a piece on each side...

File: tests/test_geometry.py [hunk 2]
```diff
@@ -201,3 +201,57 @@ def test_geobox():
...
+
+    wrap = geometry.polygon([(12231455.716333, -5559752.598333),
+                             (12231455.716333, -4447802.078667),
+                             (13343406.236, -4447802.078667),
+                             (13343406.236, -5559752.598333),
+                             (12231455.716333, -5559752.598333)], crs=sinus_crs)
+    wrapped = wrap.to_crs(geog_crs)
+    assert wrapped.type == 'Polygon'
+    wrapped = wrap.to_crs(geog_crs, wrapdateline=True)
+    # assert wrapped.type == 'MultiPolygon' TODO: implement CutGeometryOnDateLineAndAddToMulti equivalent
```
Comment (omad) [inline tests/test_geometry.py]: Are these asserts ready to be uncommented?

**Review by omad — CHANGES_REQUESTED** [review]
Cool functionality!

Should have some docs before it lands.

File: tests/test_geometry.py (hunk 2 above)
Comment (v0lat1le) [reply tests/test_geometry.py]: Nope, and probably never will be. It's quite wild to implement.
Hopefully, the GDAL method will be accessible through python API one day.
I'll add more details as a comment there I guess

**Review by omad — APPROVED** [review]

**Pull request merged**